src/sync/sync-utils.ts

```diff
@@ -18,22 +18,27 @@ export async function findOrStartSync(
 ): Promise<void> {
 	let fullSyncStartTime;
 	const { gitHubInstallationId: installationId, jiraHost } = subscription;
-	// Set sync status to PENDING, reset number of synced repos, remove repository cursor and status
 	await subscription.update({
 		syncStatus: SyncStatus.PENDING,
 		numberOfSyncedRepos: 0,
-		totalNumberOfRepos: null,
-		repositoryCursor: null,
-		repositoryStatus: null,
 		syncWarning: null
 	});
 
 	logger.info({ subscription, syncType }, "Starting sync");
 
 	if (syncType === "full") {
-		// Remove all state as we're starting anew
-		await RepoSyncState.deleteFromSubscription(subscription);
 		fullSyncStartTime = new Date().toISOString();
+		if (targetTasks && targetTasks.length > 0) {
+			await RepoSyncState.resetTasksSyncFromSubscription(targetTasks, subscription);
+		} else {
+			await subscription.update({
+				totalNumberOfRepos: null,
+				repositoryCursor: null,
+				repositoryStatus: null
+			});
+			// Remove all state as we're starting anew
+			await RepoSyncState.deleteFromSubscription(subscription);
+		}
```

Comment:
This logic isn't correct.  It's not just for a full backfill, but we want the ability to reset specific things for partial as well.

Hm... maybe we want to stop using the concept of 'full' or 'partial' and just make it "what do you want to reset before I send this backfill message to the queue"..